extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -509,6 +503,15 @@ export abstract class AzureAuth implements vscode.Disposable {
 					tenantCategory: tenantInfo.tenantCategory
 				} as Tenant;
 			});
+
+			const ignoreList = getTenantIgnoreList();
+			tenants.forEach((tenant, i) => {
+				if (ignoreList.includes(tenant.id)) {
+					Logger.verbose(`Tenant: ${tenant.id} found in ignore list, it will not be used for accessing resources.`);
+					tenants.splice(i);
```

Comment:
1. This is removing ALL items starting with index i. You need to pass in an additional parameter for the delete count : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
2. Splicing the list in-place like this will result in skipped items. See [this example](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQTpBSIYQAXlAAG0AAz8ACM-AATPwAMz8AAs-AArABdADcinkkFpAWS8DIAFEGBoABQqqWIfgZbhrACUoDpAD5QABvRQlEoUwTwdwBaDwZjqmm6iUWtLSwQBQQLdSQdU8Nbsl2KAC+weCKhwzQ8ZC8iHlzmB7BoXzcZmY7AYzEg-Fg0HUiQYwL0Ii8ACIFasGIgFvoy0lsWXwA5UAByYQAOUgEjLQRUAHkvmR3oIc1pIEdUI4FK5pHhQE54zphlh9CgEIhocJLqmJPLoiIDBxsQsmKwODwgiUgA)

I'd suggest using filter instead

Reply:
I removed that, as we don't need that for now.. will handle tenant list later as it's not trivial.